src/LesserPhp/Block.php

```diff
@@ -0,0 +1,124 @@
+<?php
+namespace LesserPhp;
+
+/**
+ * lesserphp
+ * https://www.maswaba.de/lesserphp
+ *
+ * LESS CSS compiler, adapted from http://lesscss.org
+ *
+ * Copyright 2013, Leaf Corcoran <leafot@gmail.com>
+ * Copyright 2016, Marcus Schwarz <github@maswaba.de>
+ * Copyright 2017, Stefan Pöhner <github@poe-php.de>
+ * Licensed under MIT or GPLv3, see LICENSE
+ *
+ * @package LesserPhp
+ */
+
+class Block
+{
+    /**
+     * @var Block|null
+     */
+    public $parent;
+
+    /**
+     * @var string|null
+     */
+    public $type;
+
+    /**
+     * @var int
+     */
+    public $id;
+
+    /**
+     * @var bool
+     */
+    public $isVararg = false;
+
+    /**
+     * @var array|null
+     */
+    public $tags;
+
+    /**
+     * @var array
+     */
+    public $props = [];
+
+    /**
+     * @var Block[]
+     */
+    public $children = [];
+
+    /**
+     * Position of this block.
+     *
+     * @var int
+     */
+    public $count;
+
+    /**
+     * Add a reference to the parser so
+     * we can access the parser to throw errors
+     * or retrieve the sourceName of this block.
+     *
+     * @var Parser
+     */
+    public $parser;
+
+    /**
+     * Block constructor.
+     *
+     * @param Parser     $parser
+     * @param int        $id
+     * @param int        $count
+     * @param array|null $tags
+     * @param Block|null $parent
+     */
+    public function __construct(Parser $parser, $id, $count, array $tags = null, Block $parent = null)
+    {
+        $this->parser = $parser;
+        $this->id     = $id;
+        $this->count  = $count;
+        $this->type   = $this->getType();
+        $this->tags   = $tags;
+        $this->parent = $parent;
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getType()
+    {
+        return null;
```

Comment:
This is replacing the former stdClass object which is in some places tested with isset($obj->type). Any value other than null will crash that code.

The getter is used to give child-classes a good way to override the type instead of some mumbo jumbo in the constructor.

As soon as all switch statements using the type are converted to instanceof, the complete type thing is handled by the factory.